upgrade.md

```diff
@@ -26,55 +27,67 @@ Provided you have suitable udev rules in play, this will already do everything y
 
 ### dfu-util
 
-Download or compile a Black Magic Debug (BMD) firmware.
-As with the [bmputil](#bmputil) section above, the same considerations exist with this method for firmware selection.
+Install [`dfu-util`](http://dfu-util.sourceforge.net/). On macOS you can use homebrew, macports or fink depending on your preferred package manager. You will need version 0.8.0 or greater to support the dfuse commands.
 
-Plug in the Black Magic Probe into your computer and run the following command:
+Plug the Black Magic Probe into your computer and run the following command:
 
 ```bash
 sudo dfu-util -d 1d50:6018,:6017 -s 0x08002000:leave -D blackmagic-native.bin
 ```
 
-You will need to install [`dfu-util`](http://dfu-util.sourceforge.net/) package. On MacOS you can use homebrew, macports or fink depending on your preferred package manager. You will need version 0.8.0 or greater to support the dfuse commands.
+To upgrade non-native hardware see the readme of the different [platforms on GitHub](https://github.com/blackmagic-debug/blackmagic/tree/main/src/platforms).
 
 ```{note}
 If `dfu-util` fails to switch your BMP into bootloader mode, or you feel like you might have **bricked** your BMP, you can also plug in your BMP while holding down the button. This will force the BMP to stay in the bootloader on power up.
 ```
 
+### stlink-tool
+
+An ST-Link can be used to upgrade the firmware using [stlink-tool](<https://github.com/jeanthom/stlink-tool>).
```

Comment:
`The firmware on an ST-Link can be upgraded`? Seeing the tool can only be used with the ST-Link bootloader and firmware and replaces the donge's firmware and not any target attached to the dongle's

Reply:
Thanks for correcting me on this one. I have update the description according to your suggestion and added some notes on which repository and branch to use, including why this repository and branch shall be used. I also added a note such that other users don't make the same oopsie as I did, thinking that `stlink-tool` can be used to update the firmware of a target connected to an ST-Link.

Question: given that your repository links to [sakana280's fork](https://github.com/sakana280/stlink-tool) to use stlink-tool in Windows, do you prefer to:
- keep the subsection about `stlink-tool` the same for Windows and Linux/macOS, and perhaps update [dragonmux/stlink-tool](https://github.com/dragonmux/stlink-tool/tree/feature/compilation-fixes) such that it can also be used by Windows users
- delete the `stlink-tool` subsection from the Windows section
- update the link in the Windows section to [sakana280/stlink-tool](https://github.com/sakana280/stlink-tool)

?

Reply:
We're going to have to dive deeper into the stlink-tool stuff to figure out the answer to that. We might just move our fork to the BMD org and create a new default branch that has all the outstanding fixes merged together and tell users to use that.

It'll be later this week before we get to that, but it's probably the path of least resistance

Reply:
We're going to merge this as-is as it's cheap enough to open a new PR to adjust that set of instruction after we get to fixing that. This way users get to benefit sooner, so we're going to approve the PR but leave this unresolved so we hopefully remember to do this.

Reply:
We finally got back to this so we're opening a PR to address this, and marking this resolved at last